tests/tests.py

```diff
@@ -0,0 +1,223 @@
+            [["no-"]],
+            [["{#}"]],
+            [["-ne"]],
+            [["{#}"]],
+            [["-o-", "mi-", "ni"]],
+        ]
+        partial_text_with_all_music_missing = "{cantic- #} {#} {# -ovum}"
+        exp_partial_text_with_all_music_missing = [
+            [
+                ["{cantic- #} {#} {# -ovum}"],
+            ]
+        ]
+        text_with_section_break = "quia mirabilia fecit | salvavit sibi dextera eius"
+        exp_text_with_section_break = [
+            [["qui-", "a"], ["mi-", "ra-", "bi-", "li-", "a"], ["fe-", "cit"]],
+            [["|"]],
+            [
+                ["sal-", "va-", "vit"],
+                ["si-", "bi"],
+                ["dex-", "te-", "ra"],
+                ["e-", "ius"],
+            ],
+        ]
+        text_with_incipit = "et brachium sanctum eius | ~Gloria | ~Ipsum [Canticum]"
+        exp_text_with_incipit = [
+            [["et"], ["bra-", "chi-", "um"], ["sanc-", "tum"], ["e-", "ius"]],
+            [["|"]],
+            [["~Gloria"]],
+            [["|"]],
+            [["~Ipsum [Canticum]"]],
+        ]
+        all_test_strings = [
+            normal_text,
+            missing_complete_words,
+            missing_partial_words,
+            words_with_missing_music,
+            partial_words_with_missing_music,
+            missing_whole_words_and_music,
+            missing_partial_words_and_music,
+            partial_text_with_all_music_missing,
+            text_with_section_break,
+            text_with_incipit,
+        ]
+        full_expected_result = [
+            exp_normal_text,
+            exp_missing_complete_words,
+            exp_missing_partial_words,
+            exp_words_with_missing_music,
+            exp_partial_words_with_missing_music,
+            exp_missing_whole_words_and_music,
+            exp_missing_partial_words_and_music,
+            exp_partial_text_with_all_music_missing,
+            exp_text_with_section_break,
+            exp_text_with_incipit,
+        ]
+        for test_case_num in range(10):
+            with self.subTest(test_case_num=test_case_num):
+                self.assertEqual(
+                    syllabify_text(all_test_strings[test_case_num]),
+                    full_expected_result[test_case_num],
+                )
+
+
+class TestTextVolpianoAlignment(unittest.TestCase):
+    """
+    Tests functions for aligning text and volpiano in
+    text_volpiano_alignment.py.
+    """
+
+    def test_align_text_volpiano(self):
+        """
+        Tests align_text_volpiano.
+        """
+        with open(
+            "tests/alignment_test_cases.json", encoding="ascii"
+        ) as test_case_json:
+            test_cases = json.load(test_case_json)
+        for test_case in test_cases:
+            tupled_case = []
+            for list_elem in test_case["expected_result"]:
+                tupled_case.append(tuple(list_elem))
+            test_case["expected_result"] = tupled_case
```

Comment:
I don't understand what's going on here. Looking at the first test case in `alignment_test_cases.json`, we're turning 
```python
["", "1---"],
["in", "f---"],
["di-", "df--"],
...
```
into 
```python
("", "1---"),
("in", "f---"),
("di-", "df--"),
...
```
? A comment here could be helpful explaining why this is necessary...

Reply:
oh, this is because `align_text_and_volpiano` returns a list of tuples, right?

Maybe a comment here could still be useful, but it's probably not strictly necessary.